src/ucc/api/ucc.h

```diff
@@ -1012,8 +1012,42 @@ typedef struct ucc_team_p2p_conn {
  *  @ingroup UCC_TEAM_DT
  */
 typedef enum {
-    UCC_COLLECTIVE_POST_ORDERED     = 0,
-    UCC_COLLECTIVE_POST_UNORDERED   = 1
+
+    /**
+      * When set to this value, the collective participants shall invoke the operation
+      * in the same order.
+      */
+    UCC_COLLECTIVE_POST_ORDERED             = 0,
+
+    /**
+      * When set to this value, the collective participants shall invoke the operation
+      * in any order.
+      */
+    UCC_COLLECTIVE_POST_UNORDERED           = 1,
+
+    /**
+      * When set to this value, the collective participants shall initialize the operation
+      * in the same order.
+      */
+    UCC_COLLECTIVE_INIT_ORDERED             = 2,
+
+    /**
+      * When set to this value, the collective participants shall initialize the operation
+      * in any order.
+      */
+    UCC_COLLECTIVE_INIT_UNORDERED           = 3,
+
+    /**
+      * When set to this value, the collective participants shall initialize and
+      * invoke the operation in the same order.
+      */
+    UCC_COLLECTIVE_INIT_AND_POST_ORDERED    = 4,
+
+    /**
+      * When set to this value, the collective participants shall initialize and
+      * invoke the operation in any order.
+      */
+    UCC_COLLECTIVE_INIT_AND_POST_UNORDERED  = 5
```

Comment:
Can we really support that unless it is a "tagged" collective?

Reply:
Yes. But there are edge cases, where we need tag-based collectives.